### PR TITLE
Add 3D operations for layers: conv3d, avg_pool3d, max_pool3d and conv3d_transpose

### PR DESCRIPTION
**Pull request by Kongsea**
1.Add 3D operations for layers:

conv3d
avg_pool3d
max_pool3d
conv3d_transpose

2.Replace a deprecated function in layers_test.py.

3.Add unit test for avg_pool3d and max_pool3d.

### CODE REVIEW

**Comment by xcyan** [comment]
What's the status of this PR?

**Comment by Kongsea** [comment]
@xcyan It's still awaiting the tensorflower's reviewing.

File: tensorflow/contrib/layers/python/layers/layers.py [hunk 1]
```diff
@@ -1178,6 +1234,115 @@ def convolution2d_transpose(
...
+    weights_initializer=initializers.xavier_initializer(),
+    weights_regularizer=None,
+    biases_initializer=init_ops.zeros_initializer(),
+    biases_regularizer=None,
+    reuse=None,
+    variables_collections=None,
+    outputs_collections=None,
+    trainable=True,
+    scope=None):
+  """Adds a convolution3d_transpose with an optional batch normalization layer.
```
Comment (fchollet) [inline tensorflow/contrib/layers/python/layers/layers.py]: Needs a line break after this line

File: tensorflow/contrib/layers/python/layers/layers_test.py [hunk 2]
```diff
@@ -49,6 +48,7 @@
 from tensorflow.python.ops import variable_scope
 from tensorflow.python.ops import variables as variables_lib
 from tensorflow.python.platform import test
+from tensorflow.python.ops.losses.losses import get_regularization_losses
```
Comment (fchollet) [inline tensorflow/contrib/layers/python/layers/layers_test.py]: Don't import methods, only import at module level.

**Comment by Kongsea** [comment]
Thank you. I have fixed the problems you mentioned.

**Review by fchollet — APPROVED** [review]
LGTM assuming tests pass.

**Comment by jhseu** [comment]
Some changes from elsewhere got into RELEASE.md. Mind reverting those?

**Comment by jhseu** [comment]
Jenkins, test this please

**Comment by Kongsea** [comment]
@jhseu OK, now I have reverted the changes of RELEASE.md. Please check it.

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by Kongsea** [comment]
After testing, It gives "Some checks were not successful.". When I look at the testing outputs, I found an error:

Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-gpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local_linux-opt/bin/tensorflow/contrib/layers/layers_test.runfiles/org_tensorflow/tensorflow/contrib/layers/python/layers/layers_test.py", line 51, in <module>
    from tensorflow.python.ops.losses import losses
ImportError: No module named losses

However, when I run it on my computer, or import the losses module directly, it has no error. So what's the reason? What need I do to fix it?
Thank you.

File: tensorflow/contrib/layers/python/layers/layers_test.py [hunk 3]
```diff
@@ -49,6 +48,7 @@
 from tensorflow.python.ops import variable_scope
 from tensorflow.python.ops import variables as variables_lib
 from tensorflow.python.platform import test
+from tensorflow.python.ops.losses import losses
```
Comment (martinwicke) [inline tensorflow/contrib/layers/python/layers/layers_test.py]: Can you put this in alphabetical order, i.e. switch it with the line before?

**Review by martinwicke — DISMISSED** [review]
The reason for the test failure is that in contrib/layers/BUILD, layers_test does not depend on tensorflow/python/ops/losses, so when you execute the test, it isn't given access to these files. Just add that dependency there and you should be fine.

**Comment by Kongsea** [comment]
@martinwicke Thank you.
Now, I have added the dependency for losses and rescheduled the order of the import. Please check it.

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by martinwicke** [comment]
It needs to be "tensorflow/python/ops/losses".

```
ERROR: /workspace/tensorflow/contrib/layers/BUILD:116:1: no such package 'tensorflow/python/losses': BUILD file not found on package path and referenced by '//tensorflow/contrib/layers:layers_test'.
ERROR: Analysis of target '//tensorflow/contrib/layers:layers_test' failed; build aborted.
```

**Comment by Kongsea** [comment]
@martinwicke Thank you. I have modified it. Please test it again.

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by martinwicke** [comment]
It needs to be `//tensorflow/python/ops/losses` or `//tensorflow/python/ops/losses:losses`, without the _py.

**Comment by Kongsea** [comment]
@martinwicke Sorry.
Now I have modified it. Please check and test it again.

**Comment by martinwicke** [comment]
Jenkins, test this please!

**Comment by xcyan** [comment]
Fixed #9347

**Pull request merged**